### PR TITLE
Rename fork selection and locktower to Tower BFT Consensus

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

Name the consensus algo.

#### Summary of Changes

Rename fork selection and locktower to Tower BFT.  Rename locktower to Tower.  Make metrics and all the names are consistent.  

Fixes #

### CODE REVIEW

File: core/src/cluster_info_vote_listener.rs [hunk 1]
```diff
@@ -85,8 +85,8 @@ impl Service for ClusterInfoVoteListener {
 
 #[cfg(test)]
 mod tests {
-    use crate::locktower::MAX_RECENT_VOTES;
     use crate::packet;
+    use crate::tower_bft_consensus::MAX_RECENT_VOTES;
```
Comment (garious) [inline core/src/cluster_info_vote_listener.rs]: I'm okay with calling this module `consensus.rs`. We could rename it to something more specific if we every had two consensus mechanisms living side by side, but I don't see that happening any time soon.

File: core/src/replay_stage.rs [hunk 2]
```diff
@@ -445,42 +445,42 @@ impl ReplayStage {
...
         }
-        inc_new_counter_info!("replay_stage-locktower_duration", ms as usize);
+        inc_new_counter_info!("replay_stage-tower_bft_duration", ms as usize);
 
         votable
     }
 
     fn confirm_forks(
-        locktower: &Locktower,
+        tower_bft: &TowerBFT,
```
Comment (garious) [inline core/src/replay_stage.rs]: `tower: Tower` would suffice

File: book/src/SUMMARY.md [hunk 3]
```diff
@@ -58,7 +58,7 @@
   - [Validator](validator-proposal.md)
 
 - [Implemented Design Proposals](implemented-proposals.md)
-  - [Fork Selection](fork-selection.md)
+  - [Tower BFT Consensus](tower-bft-consensus.md)
```
Comment (garious) [inline book/src/SUMMARY.md]: You can just call this `Tower BFT`

File: book/src/block-confirmation.md [hunk 4]
```diff
@@ -50,12 +50,12 @@ log the time since the NewBlock transaction was submitted.
...
-Locktower is the proposed [fork selection](fork-selection.md) algorithm. It
-proposes that payment to miners be postponed until the *stack* of validator
+[Tower BFT Consensus](tower-bft-consensus.md) is the proposed fork selection algorithm.
+It proposes that payment to miners be postponed until the *stack* of validator
 votes reaches a certain depth, at which point rollback is not economically
-feasible. The vote program may therefore implement locktower. Vote instructions
-would need to reference a global locktower account so that it can track
-cross-block state.
+feasible. The vote program may therefore implement [Tower
+Consensus](tower-bft-consensus.md). Vote instructions would need to reference a
```
Comment (garious) [inline book/src/block-confirmation.md]: You can remove the second link.

**Comment by aeyakovenko** [comment]
@sagar-solana 
any idea why CI fails with this:
```

 
--
  | failures:
  | test_replicator_startup_1_node

```

**Comment by sagar-solana** [comment]
No idea. Weird. @danpaul000 ?

**Comment by mvines** [comment]
I’ve seen that test fail occasionally too.  I think we have a number of racy tests again

**Comment by aeyakovenko** [comment]
Seems to fail repeatedly for this PR

**Comment by sagar-solana** [comment]
It's unable to bind to some crazy high port. Must be rpc or something binding to "any". Is it running on the same CI machine repeated ?

**Pull request merged**